src/server/config_registry.cc

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023, Roman Gershman.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+#include "src/server/config_registry.h"
+
+#include <absl/flags/reflection.h>
+
+#include "base/logging.h"
+
+namespace dfly {
+
+using namespace std;
+
+ConfigRegistry& ConfigRegistry::Register(std::string_view name, WriteCb cb) {
+  absl::CommandLineFlag* flag = absl::FindCommandLineFlag(name);
+  CHECK(flag) << "Unknown config name: " << name;
+
+  unique_lock lk(mu_);
+  auto [it, inserted] = registry_.emplace(name, std::move(cb));
+  CHECK(inserted) << "Duplicate config name: " << name;
+  return *this;
+}
+
+// Returns true if the value was updated.
+bool ConfigRegistry::Set(std::string_view config_name, std::string_view value) {
```

Comment:
I'd add a `_test.cc` for basic usage of `Register()`, `Set()` (success & fail) and `Reset()`.

Reply:
I am not a fan of umbrella rules like "every class needs a test". i prefer to wait for a bug happen in this code and add a test then. In other words, reaching 100% test coverage - is not a goal.

Reply:
Sure thing. Only please allow me to print this and hang it in the office :smiling_imp: